sdk/lib/opentelemetry/sdk/trace/samplers.rb

```diff
@@ -76,35 +79,27 @@ module Samplers
         #   ignore. Defaults to ignore {OpenTelemetry::Trace::SamplingHint::RECORD}.
         # @param [optional Boolean] ignore_parent Whether to ignore parent
         #   sampling. Defaults to not ignore parent sampling.
-        # @param [optional Boolean] apply_to_root_spans Whether to apply
-        #   probability sampling to root spans. Defaults to true.
-        # @param [optional Boolean] apply_to_remote_parent Whether to apply
-        #   probability sampling to remote parent. Defaults to true.
-        # @param [optional Boolean] apply_to_all_spans Whether to apply
-        #   probability sampling to all spans. Defaults to false.
+        # @param [optional Symbol] apply_probability_to Whether to apply
+        #   probability sampling to root spans, root spans and remote parents,
+        #   or all spans. Allowed values include :root_spans, :root_spans_and_remote_parent,
+        #   and :all_spans. Defaults to :root_spans_and_remote_parent.
         # @raise [ArgumentError] if probability is out of range
         # @raise [ArgumentError] if ignore_hints contains invalid hints
-        # @raise [ArgumentError] unless apply_to_all_spans implies apply_to_root_spans
-        #   and apply_to_remote_parent
-        # @return [OpenTelemetry::Trace::Samplers::Sampler]
+        # @raise [ArgumentError] if apply_probability_to is not one of the allowed symbols
         def self.probability(probability,
                              ignore_hints: [OpenTelemetry::Trace::SamplingHint::RECORD],
                              ignore_parent: false,
-                             apply_to_root_spans: true,
-                             apply_to_remote_parent: true,
-                             apply_to_all_spans: false)
+                             apply_probability_to: :root_spans_and_remote_parent)
```

Comment:
The collection of boolean flags was confusing, interdependent, incorrectly implemented, and didn't capture either the spirit or intent of the OTEP. Since we're in Ruby, we can have Symbols, so this uses a single Symbol argument to distinguish the 3 supported cases.